src/app/games/lasca/lasca.component.ts

```diff
@@ -0,0 +1,344 @@
+import { Component } from '@angular/core';
+import { RectangularGameComponent } from 'src/app/components/game-components/rectangular-game-component/RectangularGameComponent';
+import { Coord } from 'src/app/jscaip/Coord';
+import { Player } from 'src/app/jscaip/Player';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { MessageDisplayer } from 'src/app/services/MessageDisplayer';
+import { Localized } from 'src/app/utils/LocaleUtils';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MGPSet } from 'src/app/utils/MGPSet';
+import { MGPValidation } from 'src/app/utils/MGPValidation';
+import { Utils } from 'src/app/utils/utils';
+import { LascaControlAndDominationMinimax } from './LascaControlAndDomination';
+import { LascaControlMinimax } from './LascaControlMinimax';
+import { LascaMove, LascaMoveFailure } from './LascaMove';
+import { LascaRules } from './LascaRules';
+import { LascaPiece, LascaSpace, LascaState } from './LascaState';
+import { LascaTutorial } from './LascaTutorial';
+
+export class LascaComponentFailure {
+
+    public static readonly THIS_PIECE_CANNOT_MOVE: Localized = () => $localize`This piece cannot move!`;
+}
```

Comment:
eh faut se décider sur la convention lignes vides !
Ici c'est pas homogène entre LascaComponentFailure et LascaComponent

Reply:
Je dirais, la première ligne d'une classe (qu'on ne collapsera pas donc) doit être vide comme ça ça fait pas pavé quand tu lis le résumé !
Mais pas dans une fonction/méthode, et pas besoin entre les méthode car le } servira de séparateur de lignes !

Reply:
et pourquoi première ligne d'une classe et pas première ligne d'une interface ? (cf. juste en dessous)
Faut que tu me screenshot parce que j'ai du mal à voir l'aspect pavé ici (et j'ai pas ton éditeur)

Le raisonnement de séparer avec une ligne vide entre les méthodes c'était d'aider git en cas de conflit, mais on a moins de conflit donc potentiellement OK de pas séparer. Par contre cette convention de première ligne vide c'est chelou

Reply:
Voir Trello

Reply:
voir ousaaaa

Reply:
euh, je sais plus, skip it for now :p ?